src/openapi/streaming/subscription.ts

```diff
@@ -1020,8 +1044,17 @@ class Subscription {
      * can ignore them. It also ensures that we don't hit the subscription limit
      * because the subscribe manages to get to the server before the unsubscribe.
      */
-    reset() {
-        this.checkIfPublisherDown();
+    reset(isServerInitiated: boolean) {
+        this.checkIfPublisherDown(isServerInitiated);
+        this.unsubscribeAndSubscribe();
+    }
+
+    /**
+     * Does a unsubscribe and then schedules a subscribe for when it is finished unsubscribing
+     * Call this only if this is not a error scenario. Normally you should use reset
+     * So that we track the reset
+     */
+    unsubscribeAndSubscribe() {

```

Comment:
Cannot comment on unchanged lines but below log might need adjusting as it's now not inclusive to reset.
```

log.error(
                    LOG_AREA,
                    'Reset was called but subscription is in an unknown state',
                );

```

Reply:
updated